compiler/il/OMRBlock.cpp

```diff
@@ -1165,17 +1165,21 @@ static TR::SymbolReference * createSymRefForNode(TR::Compilation *comp, TR::Reso
                {
                while (valueChild->getOpCode().isArrayRef())
                   valueChild = valueChild->getFirstChild();
-
-               if (valueChild->getOpCode().isLoadVarDirect() &&
-                     valueChild->getSymbolReference()->getSymbol()->isAuto())
+               // If the node we are uncommoning is internal pointer and while iterating a child we found a node that is not 
+               // Array Reference, There are only two possibilities.
+               // 1. It is  internal poiter which was commoned means could be stored into register or on temp slot. 
+               // 2. Itself is a pointer to array object.
+               TR::SymbolReference *valueChildSymRef = valueChild->getSymbolReference();
+               if ((valueChild->getOpCode().isLoadVarDirect() && valueChildSymRef->getSymbol()->isAuto()) ||
+                     (valueChild->getOpCode().isLoadReg() && valueChildSymRef != NULL && valueChildSymRef->getSymbol()->castToAutoSymbol()->isInternalPointer()))
```

Comment:
Don't both conditions in the if statement need a null test?

ie:
```C
if ((valueChildSymRef != NULL) 
&& ((valueChild->getOpCode().isLoadVarDirect() && valueChildSymRef->getSymbol()->isAuto()) 
|| (valueChild->getOpCode().isLoadReg() &&  valueChildSymRef->getSymbol()->castToAutoSymbol()->isInternalPointer()))) {
```

Reply:
Reason I did not add Null Check for `LoadVar` node is because, I am under impression that a `LoadVar` node should be created with symbol reference and it is illegal to have a node without it. If we ever encounter that kind of node, we would any way fail with segmentation fault later in instruction selection.

Reply:
>  If we ever encounter that kind of node, we would any way fail with segmentation fault later in instruction selection.

The code would segfault here if that kind of node was ever created.  =)

Personally, I would prefer the explicit null check here as it makes it the rest of the code clearer without having to know that `LoadVar` node will never not have symbol reference. I'll leave it to the you and the compiler reviewers to determine if that's worth addressing or not

Reply:
I have put explicit NULL Check in https://github.com/eclipse/omr/pull/4535/commits/26643a8384b977787a3ad9b88cb8c24684c2f75d